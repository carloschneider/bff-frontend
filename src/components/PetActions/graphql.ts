import { gql } from '@apollo/client'

export type CheckType = {
  id: string
}

export type CheckInDataType = {
  checkIn: CheckType[]
}

export type CheckVariablesType = {
  petId: string
}

export const MUTATION_CHECKIN = gql`
  mutation CheckIn($petId: String!) {
    checkIn(petId: $petId) {
      id
    }
  }
`

export type CheckOutDataType = {
  checkIn: CheckType[]
}

export const MUTATION_CHECKOUT = gql`
  mutation CheckOut($petId: String!) {
    checkOut(petId: $petId) {
      id
    }
  }
`
