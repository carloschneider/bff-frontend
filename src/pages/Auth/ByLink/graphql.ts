import { gql } from '@apollo/client'

export type AuthByKeyType = {
  token: string
  role: string
}

export type AuthByKeyDataType = {
  authByKey: AuthByKeyType
}

export type AuthByKeyInputType = {
  key: string
}

export type AuthByKeyVariablesInputType = {
  input: AuthByKeyInputType
}

export const MUTATION_AUTH_BY_KEY = gql`
  mutation AuthByKeyInput($input: AuthByKeyInput!) {
    authByKey(input: $input) {
      token
      role
    }
  }
`
