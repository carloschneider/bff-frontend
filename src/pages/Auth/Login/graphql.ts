import { gql } from '@apollo/client'

export type AuthTutorType = {
  id: string
}

export type AuthTutorDataType = {
  authTutor: AuthTutorType
}

export type AuthTutorVariablesType = {
  email: string
  company?: string
}

export type AuthTutorVariablesInputType = {
  input: AuthTutorVariablesType
}

export const MUTATION_AUTH_TUTOR = gql`
  mutation AuthTutor($input: EmailAuthInput!) {
    authTutor(input: $input) {
      id
    }
  }
`
