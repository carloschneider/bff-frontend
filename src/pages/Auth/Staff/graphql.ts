import { gql } from '@apollo/client'

export type AuthStaffType = {
  id: string
}

export type AuthStaffDataType = {
  authStaff: AuthStaffType
}

export type AuthStaffVariablesType = {
  email: string
  company?: string
}

export type AuthStaffVariablesInputType = {
  input: AuthStaffVariablesType
}

export const MUTATION_AUTH_STAFF = gql`
  mutation AuthStaff($input: EmailAuthInput!) {
    authStaff(input: $input) {
      id
    }
  }
`
