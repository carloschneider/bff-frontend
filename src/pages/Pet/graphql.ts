import { gql } from '@apollo/client'

// Breed
export type BreedType = {
  name: string
}

// Tutor
export type TutorType = {
  id: string
  firstName: string
  lastName: string
  phoneNumber: string
}

export type PetType = {
  id: string
  name: string
  birthDate: Date
  breed: BreedType
  count?: number | null
  tutors: TutorType[]
  createdAt?: Date
}

export type PetDataType = {
  getPetById: PetType
}

export type PetVariablesType = {
  petId?: string
}

export const GET_PET_BY_ID = gql`
  query GetPetById($petId: String!) {
    getPetById(petId: $petId) {
      id
      name
      birthDate
      breed {
        name
      }
      tutors {
        id
        firstName
        lastName
        phoneNumber
      }
    }
  }
`
