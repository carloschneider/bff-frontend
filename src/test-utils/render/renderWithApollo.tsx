import { MockedProvider, MockedResponse } from '@apollo/react-testing'
import { render } from '@testing-library/react'
import { ReactNode } from 'react'

export const renderWithApollo = (
  children: ReactNode | undefined,
  mocks: ReadonlyArray<MockedResponse> = []
) => {
  return render(
    <MockedProvider
      mocks={mocks}
      addTypename={false}
      defaultOptions={{ mutate: { errorPolicy: 'all' } }}
    >
      {children}
    </MockedProvider>
  )
}
