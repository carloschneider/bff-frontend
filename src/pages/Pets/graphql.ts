import { gql } from '@apollo/client'

import { OrderEnum } from 'helpers/pagination/order'
import { PetType } from 'pages/Pet/graphql'

export type PetsDataType = {
  getAllPets: Omit<PetType, 'tutors'>[]
}

export type PaginationType = {
  order?: OrderEnum
  field?: keyof PetType
  limit: number
  offset: number
}

type PetsWhereType = {
  name?: string
}

export type PetsVariablesType = {
  where?: PetsWhereType | null
} & PaginationType

export const QUERY_GET_ALL_PETS = gql`
  query GetAllPets(
    $limit: Int
    $offset: Int
    $field: String
    $order: String
    $where: PetWhereInput
  ) {
    getAllPets(
      limit: $limit
      offset: $offset
      field: $field
      order: $order
      where: $where
    ) {
      id
      name
      birthDate
      createdAt
      breed {
        name
      }
      count
    }
  }
`
