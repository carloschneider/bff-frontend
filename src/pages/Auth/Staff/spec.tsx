import { MockedResponse } from '@apollo/react-testing'
import { act, fireEvent, screen } from '@testing-library/react'
import { GraphQLError } from 'graphql'

import { renderWithApollo } from 'test-utils/render/renderWithApollo'

import { MUTATION_AUTH } from './graphql'

import PageLoginStaff from '.'

const mockedCompany = 'fake-company'
const mockedEmail = 'test@email.com'

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    company: mockedCompany
  })
}))

describe('<PageLoginStaff />', () => {
  it('should render component', () => {
    const { container } = renderWithApollo(<PageLoginStaff />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should login', async () => {
    const apolloResponseMock: MockedResponse[] = [
      {
        request: {
          query: MUTATION_AUTH,
          variables: {
            input: {
              email: mockedEmail,
              company: mockedCompany
            }
          }
        },
        result: () => {
          return {
            data: {
              authStaff: {
                id: 'fake-id'
              }
            }
          }
        }
      }
    ]

    renderWithApollo(<PageLoginStaff />, apolloResponseMock)

    const inputEmail = await screen.getByPlaceholderText<HTMLInputElement>(
      'E-mail'
    )

    await act(async () => {
      await fireEvent.change(inputEmail, {
        target: {
          value: mockedEmail
        }
      })
    })

    expect(inputEmail.value).toBe(mockedEmail)

    const submit = screen.getByRole('button')

    await act(async () => {
      await fireEvent.click(submit)
    })

    const successMessage = screen.getByText(
      'We sent a login link to your e-mail.'
    )

    expect(successMessage).toBeInTheDocument()
  })

  it('should show error when try login with a invalid email', async () => {
    renderWithApollo(<PageLoginStaff />)

    const inputEmail = await screen.getByPlaceholderText<HTMLInputElement>(
      'E-mail'
    )

    await act(async () => {
      await fireEvent.change(inputEmail, {
        target: {
          value: 'invalid@email'
        }
      })
    })

    const submit = screen.getByRole('button')

    await act(async () => {
      await fireEvent.click(submit)
    })

    const errorMessage = screen.getByText('Please input a valid e-mail')

    expect(errorMessage).toBeInTheDocument()
  })

  it('should show error when try to submit form with email empty', async () => {
    renderWithApollo(<PageLoginStaff />)

    const inputEmail = await screen.getByPlaceholderText<HTMLInputElement>(
      'E-mail'
    )

    await act(async () => {
      await fireEvent.change(inputEmail, {
        target: {
          value: '123'
        }
      })
    })

    await act(async () => {
      await fireEvent.change(inputEmail, {
        target: {
          value: ''
        }
      })
    })

    const submit = screen.getByRole('button')

    await act(async () => {
      await fireEvent.click(submit)
    })

    const errorMessage = screen.getByText('Please input your e-mail')

    expect(errorMessage).toBeInTheDocument()
  })

  it('should return error and show notification', async () => {
    const apolloErrorResponseMock: MockedResponse[] = [
      {
        request: {
          query: MUTATION_AUTH,
          variables: {
            input: {
              email: mockedEmail,
              company: mockedCompany
            }
          }
        },
        result: {
          errors: [new GraphQLError('Mocked error message')]
        }
      }
    ]

    renderWithApollo(<PageLoginStaff />, apolloErrorResponseMock)

    const inputEmail = await screen.getByPlaceholderText<HTMLInputElement>(
      'E-mail'
    )

    await act(async () => {
      await fireEvent.change(inputEmail, {
        target: {
          value: mockedEmail
        }
      })
    })

    expect(inputEmail.value).toBe(mockedEmail)

    const submit = screen.getByRole('button')

    await act(async () => {
      await fireEvent.click(submit)
    })

    const errorNotification = screen.getByText('Mocked error message')

    expect(errorNotification).toBeInTheDocument()
  })
})
