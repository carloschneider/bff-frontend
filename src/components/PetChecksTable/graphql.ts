import { gql } from '@apollo/client'

import { PaginationType } from 'pages/Pets'

export type StaffType = {
  id: string
  firstName: string
  lastName: string
}

export type UserType = {
  id: string
  companyName: string
}

export type ResponsibleType = {
  id: string
  type: 'arrive' | 'leave'
  staff: StaffType | null
  user: UserType | null
}

export type CheckType = {
  id: string
  arrive: string
  leave: string | null
  count?: number | null
  responsibles: ResponsibleType[]
}

export type ChecksDataType = {
  getAllChecksByPetId: CheckType[] | undefined
}

type ChecksWhereType = {
  id?: string
}

export type ChecksVariablesType = {
  where?: ChecksWhereType | null
} & PaginationType

export const GET_CHECKS_BY_PET_ID = gql`
  query GetAllChecksByPetId(
    $limit: Int
    $offset: Int
    $field: String
    $order: String
    $where: CheckWhereGetAllByPetIdInput!
  ) {
    getAllChecksByPetId(
      limit: $limit
      offset: $offset
      field: $field
      order: $order
      where: $where
    ) {
      id
      arrive
      leave
      count
      responsibles {
        id
        type
        staff {
          id
          firstName
          lastName
        }
        user {
          id
          companyName
        }
      }
    }
  }
`
