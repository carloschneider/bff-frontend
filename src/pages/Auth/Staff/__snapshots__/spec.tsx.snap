// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLoginStaff /> should render component 1`] = `
<div
  class="ant-space css-dev-only-do-not-override-1n7nwfa ant-space-vertical"
>
  <div
    class="ant-space-item"
  >
    <div
      class="ant-card ant-card-bordered card css-dev-only-do-not-override-1n7nwfa"
    >
      <div
        class="ant-card-cover"
      >
        <img
          src="https://placedog.net/300?random"
        />
      </div>
      <div
        class="ant-card-body"
      >
        <h1
          class="ant-typography css-dev-only-do-not-override-1n7nwfa"
          style="font-size: 20px;"
        >
          Login
        </h1>
        <form
          class="ant-form ant-form-vertical css-dev-only-do-not-override-1n7nwfa"
        >
          <div
            class="ant-form-item css-dev-only-do-not-override-1n7nwfa"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1n7nwfa"
            >
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-1n7nwfa"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <span
                      class="ant-input-affix-wrapper css-dev-only-do-not-override-1n7nwfa"
                    >
                      <span
                        class="ant-input-prefix"
                      >
                        <span
                          aria-label="user"
                          class="anticon anticon-user login-icon"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="user"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                            />
                          </svg>
                        </span>
                      </span>
                      <input
                        aria-required="true"
                        class="ant-input css-dev-only-do-not-override-1n7nwfa"
                        id="email"
                        placeholder="E-mail"
                        type="email"
                        value=""
                      />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-space css-dev-only-do-not-override-1n7nwfa ant-space-vertical"
            style="display: flex;"
          >
            <div
              class="ant-space-item"
            >
              <div
                class="ant-form-item login-button css-dev-only-do-not-override-1n7nwfa"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-1n7nwfa"
                >
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-1n7nwfa"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <button
                          class="ant-btn css-dev-only-do-not-override-1n7nwfa ant-btn-primary"
                          type="submit"
                        >
                          <span>
                            Login
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
